packages/reports/addon/consumers/request/filter.js

```diff
@@ -51,14 +51,18 @@ export default ActionConsumer.extend({
     /**
      * @action TOGGLE_DIM_FILTER
      * @param {Object} route - route that has a model that contains a request property
+     * @param {Function} expandFilters - expand filters section action
      * @param {Object} dimension - dimension to filter
      */
-    [RequestActions.TOGGLE_DIM_FILTER]: function(route, dimension) {
+    [RequestActions.TOGGLE_DIM_FILTER]: function(route, expandFilters, dimension) {
       let filteredDimensions = get(route, 'currentModel.request.filters'),
         filter = filteredDimensions.findBy('dimension', dimension);
 
       //do not add filter if it already exists
       if (!filter) {
+        if (typeof expandFilters === 'function') {
+          expandFilters();
```

Comment:
Is this expand filters function a visual thing? If that's the case, we should probably keep that out of the consumers as they are generally used for updating models and fragments rather than component properties. Perhaps a better option would be to queue an action that expands the filter collection followed by the consumer action.

Reply:
yeah, I didn't like it in the consumers as well, even though checking for the dimension existence is already there. I'll move it somewhere else